src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1286,15 +1286,19 @@ internal static FunctionGroup FirstFunctionGroupInInheritanceChain(string method
         private List<FunctionEndPoint> GetCandidateFunctions(StackFrame stackFrame,
```

Comment:
do we actually use this somewhere? How do we get stackframes?

Reply:
I made these changes because I found the conditions very confusing.
I hope my changes makes it a bit clearer